cluster/v1alpha1/types.go

```diff
@@ -301,32 +301,79 @@ const (
 
 // PrioritizerConfig represents the configuration of prioritizer
 type PrioritizerConfig struct {
+	// Name will be removed in v1beta1 and replaced by PrioritizerScoreCoordinate.BuiltIn.
+	// If both Name and PrioritizerScoreCoordinate.BuiltIn are defined, will use the value
+	// in PrioritizerScoreCoordinate.BuiltIn.
 	// Name is the name of a prioritizer. Below are the valid names:
 	// 1) Balance: balance the decisions among the clusters.
 	// 2) Steady: ensure the existing decision is stabilized.
 	// 3) ResourceAllocatableCPU & ResourceAllocatableMemory: sort clusters based on the allocatable.
-	// +kubebuilder:validation:Required
-	// +required
-	Name string `json:"name"`
+	// +optional
+	Name string `json:"name,omitempty"`
 
-	// Weight defines the weight of prioritizer. The value must be ranged in [0,10].
+	// PrioritizerScoreCoordinate represents the configuration of the prioritizer and score source.
+	// +optional
+	PrioritizerScoreCoordinate *PrioritizerScoreCoordinate `json:"scoreCoordinate,omitempty"`
+
+	// Weight defines the weight of the prioritizer score. The value must be ranged in [-10,10].
 	// Each prioritizer will calculate an integer score of a cluster in the range of [-100, 100].
 	// The final score of a cluster will be sum(weight * prioritizer_score).
 	// A higher weight indicates that the prioritizer weights more in the cluster selection,
-	// while 0 weight indicate thats the prioritizer is disabled.
-	// +kubebuilder:validation:Minimum:=0
+	// while 0 weight indicates that the prioritizer is disabled. A negative weight indicates
+	// wants to select the last ones.
+	// +kubebuilder:validation:Minimum:=-10
 	// +kubebuilder:validation:Maximum:=10
 	// +kubebuilder:default:=1
 	// +optional
 	Weight int32 `json:"weight,omitempty"`
 }
 
+// PrioritizerScoreCoordinate represents the configuration of the prioritizer and score source
+type PrioritizerScoreCoordinate struct {
```

Comment:
can we change this to `ScoreCoordinate`? I feel we might have to use it for filter in the future also.

Reply:
Update to ScoreCoordinate.